subprojects/dependency-management/src/testFixtures/groovy/org/gradle/containers/GradleContainer.groovy

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.containers
+
+
+import com.github.dockerjava.api.command.InspectContainerResponse
+import com.github.dockerjava.api.exception.DockerException
+import com.google.common.base.Charsets
+import groovy.transform.CompileStatic
+import org.testcontainers.DockerClientFactory
+import org.testcontainers.containers.GenericContainer
+import org.testcontainers.containers.output.FrameConsumerResultCallback
+import org.testcontainers.containers.output.OutputFrame
+import org.testcontainers.containers.output.ToStringConsumer
+import org.testcontainers.utility.TestEnvironment
+
+@CompileStatic
+class GradleContainer extends GenericContainer<GradleContainer> {
+    GradleContainer(String image) {
+        super(image)
+    }
+
+    @Override
+    void close() {
+        stop()
+    }
+
+    // this code is mostly copied from ExecInContainerPattern in order to be able to
+    // follow Gradle execution live
+    GradleExecResult execute(String... command) {
+        if (!TestEnvironment.dockerExecutionDriverSupportsExec()) {
```

Comment:
this check is rather obsolete now and can be safely removed

Reply:
Got it, thanks.